tests/form.js

```diff
@@ -112,39 +113,16 @@ define([
             chai.expect(label.p.labelItextID.defaultValue()).to.include("/data/hidden");
         });
 
-        it("should merge data-only nodes with control nodes", function () {
-            var form = new Form({}),
-                values = [];
-
-            form.dataTree = makeTree("data", ["a", "x1", "b", "x2", "c"]);
-            form.controlTree = makeTree("control", ["a", "b", "c"]);
-
-            form.mergedTreeMap(function (v) { values.push(v.id); });
-            assert.equal(values.join(" "), "a x1 b x2 c");
-        });
```

Comment:
After a bit more reflection, I think it would be good to rewrite this test as well. I'm not sure if there are any other tests that verify that the order of hidden values interspersed with other questions is preserved.